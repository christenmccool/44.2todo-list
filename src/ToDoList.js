import React from 'react';
import './ToDoList.css';
import ToDo from './ToDo';

const ToDoList = ({todos, toggleComplete, editToDo, removeToDo}) => {
  return (
    <div className="ToDoList">
      {todos.map(todo => <ToDo 
                            key={todo.id} 
                            id={todo.id} 
                            task={todo.task} 
                            priority={todo.priority} 
                            complete={todo.complete} 
                            toggleComplete={() => toggleComplete(todo.id)}
                            editToDo={editToDo}
                            removeToDo={() => removeToDo(todo.id) } 
                          />)}
    </div>
  )
}

export default ToDoList;