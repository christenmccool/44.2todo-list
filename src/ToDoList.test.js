import { render, asFragment } from '@testing-library/react';
import ToDoList from './ToDoList';

let todos;

beforeEach(() => {
  todos = [{key:1, task: "test task", priority: "medium"}, {key:2, task: "another test task", priority: "low"}];
})

test('renders without crashing', () => {
  render(<ToDoList todos={todos}/>);
});

test('matches snapshot', () => {
    const {asFragment} = render(<ToDoList todos={todos}/>);
    expect(asFragment()).toMatchSnapshot();
});

test('renders todos', () => {
  const {queryByText, debug} = render(<ToDoList todos={todos}/>);
  expect(queryByText('test task')).toBeInTheDocument();
  expect(queryByText('another test task')).toBeInTheDocument();
  // debug();
});
