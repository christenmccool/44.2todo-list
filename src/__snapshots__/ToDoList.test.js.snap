// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ToDoList"
  >
    <div
      class="ToDo"
    >
      <span
        class="ToDo ToDo-medium"
      >
        test task
      </span>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-trash fa-w-14 ToDo-icon"
        data-icon="trash"
        data-prefix="fas"
        data-testid="trash"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-pen fa-w-16 ToDo-icon"
        data-icon="pen"
        data-prefix="fas"
        data-testid="pen"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="ToDo"
    >
      <span
        class="ToDo ToDo-low"
      >
        another test task
      </span>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-trash fa-w-14 ToDo-icon"
        data-icon="trash"
        data-prefix="fas"
        data-testid="trash"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-pen fa-w-16 ToDo-icon"
        data-icon="pen"
        data-prefix="fas"
        data-testid="pen"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
